### DISCUSSION TITLE
📚 Custom Nested Element does not work with secondary repeater control

### DISCUSSION THREAD

**Discussion by webdevs-pro** [elementor/elementor#25484] [Developers Community]
### Prerequisites

- [X] I have searched for similar issues in open and closed tickets and cannot find a duplicate.
- [X] I have troubleshooted my issue, and it still exists against the latest stable version of Elementor.

### Description

I am encountering an issue while implementing a custom widget that utilizes nested elements. I have added a new repeater control using Controls_Manager::REPEATER. However, when I click the 'Add Item' button on this repeater control, a new nested container is created, mirroring the behavior I observe when clicking the 'Add Item' button on the main repeater Control_Nested_Repeater::CONTROL_TYPE. The same issue occurs when removing an item.

![Recording 2024-03-06 132600](https://github.com/elementor/elementor/assets/14045233/cb9a0c32-f363-4dc1-b944-07bed0906eac)

Here is my code:
```
$this->start_controls_section( 'section_tabs', [
	'label' => 'Tabs',
	'tab'   => Controls_Manager::TAB_LAYOUT,
] );

$tabs_repeater = new Repeater();

$tabs_repeater->add_control( 'tab_title', [
	'label'       => 'Tab Title',
	'label_block' => true,
	'type'        => Controls_Manager::TEXT,
	'default'     => 'Tab Title',
	'placeholder' => 'Tab Title',
	'ai'          => [
		'active' => false,
	],
] );

$this->add_control( 'tabs', [
	'show_label'  => false,
	'type'        => Control_Nested_Repeater::CONTROL_TYPE,
	'fields'      => $tabs_repeater->get_controls(),
	'title_field' => '{{{ tab_title }}}',
	'button_text' => 'Add Tab',
	'default'     => [
		[
			'tab_title' => esc_html__( 'Tab #1', 'elementor-pro' ),
		],
	],
] );

$this->end_controls_section();        


$this->start_controls_section( 'section_conditions', [
	'label' => 'Conditions',
	'tab'   => Controls_Manager::TAB_LAYOUT,
] );

$conditions_repeater = new Repeater();

$conditions_repeater->add_control( 'condition_title', [
	'label'       => 'Condition Title',
	'label_block' => true,
	'type'        => Controls_Manager::TEXT,
	'default'     => 'Condition Title',
	'placeholder' => 'Condition Title',
	'ai'          => [
		'active' => false,
	],
] );

$this->add_control( 'conditions', [
	'show_label'  => false,
	'type'        => Controls_Manager::REPEATER,
	'fields'      => $conditions_repeater->get_controls(),
	'title_field' => '{{{ condition_title }}}',
	'button_text' => 'Add Condition',
	'prevent_empty' => false,
] );

$this->end_controls_section(); 
```

### Steps to reproduce

1. Create new nested elements widget.
2. Add new repeater control.

### Expected behavior

The repeater must act individually.

### Isolating the problem

- [ ] This bug happens when only the Elementor (and Elementor Pro) plugins are active.
- [ ] This bug happens with the Hello Elementor theme active.
- [X] I can reproduce this bug consistently by following the steps I described above.

### Elementor System Info

<details>
<summary>Click to reveal</summary>

````txt
== Server Environment ==
	Operating System: WINNT
	Software: nginx/1.16.0
	MySQL version: MySQL Community Server - GPL v8.0.16
	PHP Version: 8.1.23
	PHP Memory Limit: 256M
	PHP Max Input Vars: 4000
	PHP Max Post Size: 1000M
	GD Installed: Yes
	ZIP Installed: Yes
	Write Permissions: All right
	Elementor Library: Connected

== WordPress Environment ==
	Version: 6.4.3
	Site URL: http://dev.local
	Home URL: http://dev.local
	WP Multisite: No
	Max Upload Size: 300 MB
	Memory limit: 40M
	Max Memory limit: 256M
	Permalink Structure: /%postname%/
	Language: en-US
	Timezone: 0
	Debug Mode: Active

== Theme ==
	Name: Hello Elementor
	Version: 3.0.1
	Author: Elementor Team
	Child Theme: No

== User ==
	Role: administrator
	WP Profile lang: en_US
	User Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:123.0) Gecko/20100101 Firefox/123.0

== Active Plugins ==
	Elementor
		Version: 3.19.4
		Author: Elementor.com

	Elementor Pro
		Version: 3.19.2
		Author: Elementor.com

	Form Builder for Elementor
		Version: 0.1
		Author: Alex Ishchenko


== Features ==
	Custom Fonts: 0
	Custom Icons: 0

== Integrations ==
	


== Elementor Experiments ==
	Improved Asset Loading: Active by default
	Improved CSS Loading: Active by default
	Inline Font Icons: Active by default
	Additional Custom Breakpoints: Active by default
	admin_menu_rearrangement: Inactive by default
	Flexbox Container: Active by default
	Upgrade Swiper Library: Active by default
	Grid Container: Active
	Default to New Theme Builder: Inactive
	Hello Theme Header & Footer: Inactive
	Editor Top Bar: Inactive by default
	Optimized Gutenberg Loading: Active by default
	Build with AI: Inactive
	Landing Pages: Inactive
	Nested Elements: Active
	Lazy Load Background Images: Inactive by default
	Optimize Image Loading: Active by default
	Global Style Guide: Active by default
	Page Transitions: Inactive
	Notes: Inactive
	Display Conditions: Active
	Form Submissions: Active by default
	Scroll Snap: Inactive
	Menu: Inactive by default
	Taxonomy Filter: Inactive by default


== Log ==
	


== Elementor - Compatibility Tag ==
	
	Elementor Pro: Compatible
	Form Builder for Elementor: Compatibility not specified

== Elementor Pro - Compatibility Tag ==

````

</details>

### Agreement

- [X] I agree that my issue may be closed without action if it doesn't meet all the requirements.

**Comment by nicholaszein** [comment]
Hello @webdevs-pro!

Thank you for reaching out! 🙏 

At this moment Elementor does not provide any official API to use to create Custom Nested Elements, and we do not recommend 3rd-party developers to try and create their own. As you discovered yourself, Nested Elements allow only one repeater control to be used, otherwise you will encounter the problem you've encountered.

For the time being we have no plans to change this behavior.

We also can't guide you on how to properly implement your custom code. For that you need to check the [Developers Documentation](https://developers.elementor.com/docs).

I am migrating this issue to the [Developers Community](https://github.com/orgs/elementor/discussions/categories/developers-community), maybe another developer can help you.

Kind regards

**Reply by rayhanwealcoder** [reply]
Hi @nicholaszein Could you give me update about official API  Nested Elements. I need to develop Slider widget using nested elements .
Hope it will save me my time .
